monkeytype/typing.py

```diff
@@ -501,6 +502,64 @@ def rewrite_Generator(self, typ):
         return typ
 
 
+class RewriteMostCommonBase(TypeRewriter):
+    """
+    Relaces a union of classes by the least common
+    base of its members, provided that
+    parent classes of members are unique leading
+    up to the `object` class (no double inheritance)
+    """
+
+    def compute_bases(self, klass):
+        """
+        Compute list of bases of a given class,
+        starting from (but exluding) the root `object`,
+        ending  (and including) the actual class.
+        """
+        bases = []
+
+        curr_klass = klass
+
+        while curr_klass is not object:
+            bases.append(curr_klass)
+
+            curr_bases = curr_klass.__bases__
+
+            if len(curr_bases) != 1:
+                break
+
+            curr_base = curr_bases[0]
+            curr_klass = curr_base
+        return bases[::-1]
```

Comment:
I think what you want here is just `klass.mro()[1::-1]`? That would also handle multiple-inheritance, which you don't handle here.

Reply:
Maybe also add some tests covering multiple inheritance cases.